### PR TITLE
adding 1st helpers functions that allow for more concise test

### PR DESCRIPTION
**Pull request by roberthutto**

### CODE REVIEW

File: tf-build-tools/tf-test-groovy/src/main/groovy/com/deliveredtechnologies/terraform/helpers/Http.groovy [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import net.jodah.failsafe.RetryPolicy
+import net.jodah.failsafe.function.CheckedSupplier
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+import java.time.Duration
+import java.util.function.Consumer
+
+class Http {
+
```
Comment (Clayton7510) [inline tf-build-tools/tf-test-groovy/src/main/groovy/com/deliveredtechnologies/terraform/helpers/Http.groovy:13]: For helper classes, you should make the default constructor private, explicitly ensuring that an object cannot be created from the class (https://rules.sonarsource.com/java/tag/design/RSPEC-1118).

I was trying to think of a more elegant way to this without having everything static. And from a user interface perspective, nothing really seemed to make as much sense for what you are trying to do. Seems like maybe utility class may be the way to go.

**Review by Clayton7510 — CHANGES_REQUESTED** [review]
Looks pretty good! I had a few comments, tho. Please respond or address them. 
We should be able to get this thing released before Monday.

File: tf-build-tools/tf-cmd-api/src/test/java/com/deliveredtechnologies/terraform/api/TerraformApplyTest.java [hunk 2]
```diff
@@ -67,7 +67,6 @@ public void terraformApplyExecutesWhenAllPossiblePropertiesArePassed() throws IO
     Mockito.verify(this.executable, Mockito.times(1)).execute(captor.capture());
     String tfCommand = captor.getValue();
 
-    Assert.assertThat(tfCommand, CoreMatchers.startsWith("terraform apply"));
```
[the remark this answers is not in the document]
Comment (roberthutto) [reply tf-build-tools/tf-cmd-api/src/test/java/com/deliveredtechnologies/terraform/api/TerraformApplyTest.java:70]: yes. the terraform apply class doesn't even implement this functionality its the command line decorator.

File: tf-build-tools/tf-test-groovy/pom.xml [hunk 3]
```diff
@@ -54,12 +54,35 @@
       <version>2.0-M2-groovy-3.0</version>
       <scope>test</scope>
     </dependency>
+    <!-- https://mvnrepository.com/artifact/net.bytebuddy/byte-buddy -->
+    <dependency>
```
[the remark this answers is not in the document]
Comment (roberthutto) [reply tf-build-tools/tf-test-groovy/pom.xml:58]: oh yea it its being used

File: tf-build-tools/tf-test-groovy/src/main/groovy/com/deliveredtechnologies/terraform/helpers/Http.groovy (hunk 1 above)
Comment (roberthutto) [reply tf-build-tools/tf-test-groovy/src/main/groovy/com/deliveredtechnologies/terraform/helpers/Http.groovy:13]: yea ill add a private constructor

File: tf-build-tools/tf-test-groovy/pom.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Clayton7510) [reply tf-build-tools/tf-test-groovy/pom.xml:58]: ok, just wanted to make sure it was still needed

**Pull request merged**